bindings/node/lib/clientEncryption.js

```diff
@@ -549,6 +554,64 @@ module.exports = function (modules) {
       return value;
     }
 
+    /**
+     * Creates a collection that has encrypted document fields
+     *
+     * @template {TSchema} - Schema for the collection being created
+     * @param {Db} db - A database to create the collection in
+     * @param {string} name - The name of the collection to be created
+     * @param {object} options - Options for createDataKey and for createCollection.
+     * @param {string} options.provider - provider name
+     * @param {ClientEncryptionCreateDataKeyProviderOptions} options.createDataKeyOptions - options to pass to createDataKey
+     * @param {CreateCollectionOptions} options.createCollectionOptions - options to pass to createCollection, must include `encryptedFields`
+     * @returns {Promise<{ collection: Collection<TSchema>, encryptedFields: Document }>} - created collection and generated encryptedFields
+     * @throws {MongoCryptCreateEncryptedCollectionError} - If part way through the process createDataKey fails, an error will be rejected that has the `encryptedFields` that were created.
+     */
+    async createEncryptedCollection(db, name, options) {
+      const {
+        provider,
+        createDataKeyOptions = {},
+        createCollectionOptions: {
+          encryptedFields: { ...encryptedFields },
+          ...createCollectionOptions
+        }
+      } = options;
+
+      if (Array.isArray(encryptedFields.fields)) {
+        const createDataKeyPromises = [...encryptedFields.fields].map(async field =>
+          typeof field !== 'object' || field.keyId != null
+            ? field
+            : { ...field, keyId: await this.createDataKey(provider, createDataKeyOptions) }
+        );
+
+        const createDataKeyResolutions = await Promise.allSettled(createDataKeyPromises);
+
+        encryptedFields.fields = createDataKeyResolutions.map((resolution, index) =>
+          resolution.status === 'fulfilled' ? resolution.value : encryptedFields.fields[index]
+        );
+
+        const wereErrors = createDataKeyResolutions.some(
+          resolution => resolution.status === 'rejected'
+        );
+        if (wereErrors) {
+          const errors = createDataKeyResolutions.map(resolution =>
+            resolution.status === 'rejected' ? resolution.reason : null
+          );
+          throw new common.MongoCryptCreateEncryptedCollectionError(
+            'Could not complete creating data keys',
+            { encryptedFields, errors }
+          );
+        }
+      }
```

Comment:
This would loose the error's positional information w.r.t the data key that we failed to create. So we either need to:
- build up an object with numeric properties that map to an error
- create an array that is the same length as the fields input and put null where there was no errors
- modify this suggestion above slightly to map to a tuple/struct `{index, error}`